In [19]:
import pandas as pd


weather = pd.read_csv("../Data/Weather_Data.csv")

In [20]:
df=weather

In [21]:
# Convertir la colonne "Date/Time" en format de date/heure
df['Date/Time'] = pd.to_datetime(df['Date/Time'], format='%m/%d/%Y %H:%M', errors='coerce')

# Fonction pour remplacer l'année
def replace_year(date):
    if pd.notnull(date) and date.year == 2012:
        try:
            return date.replace(year=2023)
        except ValueError:
            pass
    return date

# Appliquer la fonction à la colonne "Date/Time"
df['Date/Time'] = df['Date/Time'].apply(replace_year)

# Afficher les dix premières lignes pour vérification
print(df.head(8000))

               Date/Time  Temp_C  Dew Point Temp_C  Rel Hum_%  \
0    2023-01-01 00:00:00    -1.8              -3.9         86   
1    2023-01-01 01:00:00    -1.8              -3.7         87   
2    2023-01-01 02:00:00    -1.8              -3.4         89   
3    2023-01-01 03:00:00    -1.5              -3.2         88   
4    2023-01-01 04:00:00    -1.5              -3.3         88   
...                  ...     ...               ...        ...   
7995 2023-11-29 03:00:00    -7.4             -12.2         68   
7996 2023-11-29 04:00:00    -8.4             -12.4         73   
7997 2023-11-29 05:00:00    -8.4             -11.7         77   
7998 2023-11-29 06:00:00    -6.0             -10.2         72   
7999 2023-11-29 07:00:00    -5.0             -10.6         65   

      Wind Speed_km/h  Visibility_km  Press_kPa               Weather  
0                   4            8.0     101.24                   Fog  
1                   4            8.0     101.24                   Fog  
2  

In [22]:
df.drop(columns=["Dew Point Temp_C"], inplace=True)
df

Date/Time  Temp_C  Rel Hum_%  Wind Speed_km/h  Visibility_km  \
0    2023-01-01 00:00:00    -1.8         86                4            8.0   
1    2023-01-01 01:00:00    -1.8         87                4            8.0   
2    2023-01-01 02:00:00    -1.8         89                7            4.0   
3    2023-01-01 03:00:00    -1.5         88                6            4.0   
4    2023-01-01 04:00:00    -1.5         88                7            4.8   
...                  ...     ...        ...              ...            ...   
8779 2023-12-31 19:00:00     0.1         81               30            9.7   
8780 2023-12-31 20:00:00     0.2         83               24            9.7   
8781 2023-12-31 21:00:00    -0.5         93               28            4.8   
8782 2023-12-31 22:00:00    -0.2         89               28            9.7   
8783 2023-12-31 23:00:00     0.0         86               30           11.3   

      Press_kPa               Weather  
0        101.24                   Fog  
1        101.24                   Fog  
2        101.26  Freezing Drizzle,Fog  
3        101.27  Freezing Drizzle,Fog  
4        101.23                   Fog  
...         ...                   ...  
8779     100.13                  Snow  
8780     100.03                  Snow  
8781      99.95                  Snow  
8782      99.91                  Snow  
8783      99.89                  Snow  

[8784 rows x 7 columns]

In [23]:
# Vérifier à nouveau les valeurs nulles dans le DataFrame
null_values = df.isnull().sum()

# Afficher le nombre de valeurs nulles pour chaque colonne
print(null_values)


Date/Time          0
Temp_C             0
Rel Hum_%          0
Wind Speed_km/h    0
Visibility_km      0
Press_kPa          0
Weather            0
dtype: int64


In [24]:
df["target_tmp"] = df.shift(-1)["Temp_C"]


In [25]:
df = df.ffill()


In [26]:
import pandas as pd

# Supposons que "df" est votre DataFrame

# Obtenir les valeurs uniques de la colonne "Weather"
weather_uniques = df['Weather'].unique()

# Afficher les valeurs uniques
print(weather_uniques)


['Fog' 'Freezing Drizzle,Fog' 'Mostly Cloudy' 'Cloudy' 'Rain'
 'Rain Showers' 'Mainly Clear' 'Snow Showers' 'Snow' 'Clear'
 'Freezing Rain,Fog' 'Freezing Rain' 'Freezing Drizzle' 'Rain,Snow'
 'Moderate Snow' 'Freezing Drizzle,Snow' 'Freezing Rain,Snow Grains'
 'Snow,Blowing Snow' 'Freezing Fog' 'Haze' 'Rain,Fog' 'Drizzle,Fog'
 'Drizzle' 'Freezing Drizzle,Haze' 'Freezing Rain,Haze' 'Snow,Haze'
 'Snow,Fog' 'Snow,Ice Pellets' 'Rain,Haze' 'Thunderstorms,Rain'
 'Thunderstorms,Rain Showers' 'Thunderstorms,Heavy Rain Showers'
 'Thunderstorms,Rain Showers,Fog' 'Thunderstorms' 'Thunderstorms,Rain,Fog'
 'Thunderstorms,Moderate Rain Showers,Fog' 'Rain Showers,Fog'
 'Rain Showers,Snow Showers' 'Snow Pellets' 'Rain,Snow,Fog'
 'Moderate Rain,Fog' 'Freezing Rain,Ice Pellets,Fog'
 'Drizzle,Ice Pellets,Fog' 'Drizzle,Snow' 'Rain,Ice Pellets'
 'Drizzle,Snow,Fog' 'Rain,Snow Grains' 'Rain,Snow,Ice Pellets'
 'Snow Showers,Fog' 'Moderate Snow,Blowing Snow']


In [27]:
from sklearn.preprocessing import LabelEncoder
import pandas as pd

# Supposons que "df" est votre DataFrame

# Créer une instance de LabelEncoder
label_encoder = LabelEncoder()

# Appliquer l'encodage sur la colonne "Weather"
df['Weather'] = label_encoder.fit_transform(df['Weather'])

# Afficher les 10 premières lignes pour vérification
print(df.head(10))


            Date/Time  Temp_C  Rel Hum_%  Wind Speed_km/h  Visibility_km  \
0 2023-01-01 00:00:00    -1.8         86                4            8.0   
1 2023-01-01 01:00:00    -1.8         87                4            8.0   
2 2023-01-01 02:00:00    -1.8         89                7            4.0   
3 2023-01-01 03:00:00    -1.5         88                6            4.0   
4 2023-01-01 04:00:00    -1.5         88                7            4.8   
5 2023-01-01 05:00:00    -1.4         87                9            6.4   
6 2023-01-01 06:00:00    -1.5         89                7            6.4   
7 2023-01-01 07:00:00    -1.4         85                7            8.0   
8 2023-01-01 08:00:00    -1.4         85                9            8.0   
9 2023-01-01 09:00:00    -1.3         88               15            4.0   

   Press_kPa  Weather  target_tmp  
0     101.24        7        -1.8  
1     101.24        7        -1.8  
2     101.26        9        -1.5  
3     101.27       

In [28]:
df["target_hum"] = df.shift(-1)["Rel Hum_%"]


In [29]:
df = df.ffill()


In [30]:
df["target_vent"] = df.shift(-1)["Wind Speed_km/h"]


In [31]:
df.drop(columns=['Press_kPa'], inplace=True)


In [32]:
df = df.ffill()


In [33]:
df.drop(columns=['Visibility_km'] ,inplace=True)


In [34]:
df.drop(columns=['Weather'] ,inplace=True)


In [35]:
df 

Date/Time  Temp_C  Rel Hum_%  Wind Speed_km/h  target_tmp  \
0    2023-01-01 00:00:00    -1.8         86                4        -1.8   
1    2023-01-01 01:00:00    -1.8         87                4        -1.8   
2    2023-01-01 02:00:00    -1.8         89                7        -1.5   
3    2023-01-01 03:00:00    -1.5         88                6        -1.5   
4    2023-01-01 04:00:00    -1.5         88                7        -1.4   
...                  ...     ...        ...              ...         ...   
8779 2023-12-31 19:00:00     0.1         81               30         0.2   
8780 2023-12-31 20:00:00     0.2         83               24        -0.5   
8781 2023-12-31 21:00:00    -0.5         93               28        -0.2   
8782 2023-12-31 22:00:00    -0.2         89               28         0.0   
8783 2023-12-31 23:00:00     0.0         86               30         0.0   

      target_hum  target_vent  
0           87.0          4.0  
1           89.0          7.0  
2           88.0          6.0  
3           88.0          7.0  
4           87.0          9.0  
...          ...          ...  
8779        83.0         24.0  
8780        93.0         28.0  
8781        89.0         28.0  
8782        86.0         30.0  
8783        86.0         30.0  

[8784 rows x 7 columns]

## PREDICTION TEMPERATURE 
### linear regression

In [36]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score


# Charger les données
X_train = df.drop(['Date/Time','target_vent','target_hum','target_tmp'], axis=1).iloc[:7991]  # Premières 8771 lignes pour l'entraînement
y_train = df['target_tmp'].iloc[:7991]

X_test = df.drop(['Date/Time','target_vent','target_hum','target_tmp'], axis=1).iloc[-768:]  # Dernières 12 lignes pour le test
y_test = df['target_tmp'].iloc[-768:]
# Choisissez le modèle de régression (dans cet exemple, une régression linéaire)
model = LinearRegression()

# Entraînez le modèle sur les données d'entraînement
model.fit(X_train, y_train)

# Faites des prédictions sur les données de test
predictions = model.predict(X_test)

mse = mean_squared_error(y_test, predictions)
rmse = mse ** 0.5  # Root Mean Squared Error
mae = mean_absolute_error(y_test, predictions)
r2 = r2_score(y_test, predictions)

print('Mean Squared Error:', mse)
print('Root Mean Squared Error:', rmse)
print('Mean Absolute Error:', mae)
print('R² Score:', r2)

Mean Squared Error: 0.4621287260885267
Root Mean Squared Error: 0.6798005046250898
Mean Absolute Error: 0.48899725640107944
R² Score: 0.9857230498323407


 Voici une analyse des résultats que on a  obtenu:

Mean Squared Error (MSE): 0.4621287260885256

Cette valeur indique la moyenne des carrés des erreurs entre les prédictions et les valeurs réelles. Une MSE plus faible est généralement meilleure, mais pour savoir si 0.462 est bon, il faut comparer avec une baseline ou un modèle simple, ou connaître l'échelle de votre variable cible. Si la valeur de y varie de 0 à 1, par exemple, cela pourrait être assez bon.

Root Mean Squared Error (RMSE): 0.679800504625089

La RMSE est la racine carrée de la MSE et se mesure dans la même unité que la variable cible. Comme pour la MSE, une RMSE plus faible est meilleure. À nouveau, la qualité de cette valeur dépend de l'échelle de y. Si y est dans une plage étroite, un RMSE de 0.68 pourrait être acceptable.

Mean Absolute Error (MAE): 0.48899725640107844

Le MAE représente la moyenne des erreurs absolues. C'est souvent plus intuitif à interpréter que la MSE ou la RMSE. Comme les autres erreurs, une valeur plus faible est souhaitable. La qualité de ce score dépend aussi de l'échelle de votre variable cible.

R² Score: 0.9857230498323407

L'R² score, ou coefficient de détermination, indique la proportion de la variance de y qui est expliquée par le modèle. Une valeur de 0.9857 est très élevée, suggérant que le modèle explique 98.57% de la variance des données. Cela indique une très bonne performance du modèle.

In [37]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error


# Charger les données
X_train = df.drop(['Date/Time','target_vent','target_hum','target_tmp'], axis=1).iloc[:8759]  # Premières 8771 lignes pour l'entraînement
y_train = df['target_tmp'].iloc[:8759]

X_test = df.drop(['Date/Time','target_vent','target_hum','target_tmp'], axis=1).iloc[-24:]  # Dernières 12 lignes pour le test
y_test = df['target_tmp'].iloc[-24:]
# Choisissez le modèle de régression (dans cet exemple, une régression linéaire)
model = LinearRegression()

# Entraînez le modèle sur les données d'entraînement
model.fit(X_train, y_train)

# Faites des prédictions sur les données de test
predictions = model.predict(X_test)



In [38]:
import pandas as pd

# Faites des prédictions sur les données de test
predictions = model.predict(X_test)

# Arrondir les prédictions et les valeurs réelles à une seule décimale
predictions_rounded = [round(pred, 1) for pred in predictions]
y_test_rounded = [round(real, 1) for real in y_test]

# Afficher les prédictions et les valeurs réelles dans un DataFrame
predictions_df = pd.DataFrame({'Prédictions': predictions_rounded, 'Température Réelle': y_test_rounded})
print(predictions_df)


    Prédictions  Température Réelle
0         -11.1               -10.7
1         -10.6               -10.1
2          -9.9               -11.8
3         -11.6               -10.5
4         -10.3               -10.2
5         -10.0                -9.7
6          -9.5                -9.3
7          -9.0                -8.6
8          -8.3                -8.1
9          -7.8                -7.4
10         -7.1                -6.7
11         -6.5                -5.8
12         -5.5                -4.6
13         -4.4                -3.4
14         -3.2                -2.3
15         -2.1                -1.4
16         -1.3                -1.1
17         -1.0                -1.3
18         -1.2                 0.1
19          0.1                 0.2
20          0.2                -0.5
21         -0.4                -0.2
22         -0.2                 0.0
23          0.0                 0.0


### XGBOOST 

In [39]:
from sklearn.metrics import mean_absolute_error, r2_score
from sklearn.ensemble import GradientBoostingRegressor


# Choisissez le modèle de régression (dans cet exemple, une régression linéaire)
model_xgb =GradientBoostingRegressor()

# Entraînez le modèle sur les données d'entraînement
model_xgb.fit(X_train, y_train)

# Faites des prédictions sur les données de test
predictions_xgb = model_xgb.predict(X_test)

# Évaluez les performances du modèle
mse_xgb = mean_squared_error(y_test, predictions_xgb)
rmse_xgb = mse ** 0.5  # Root Mean Squared Error
mae_xgb = mean_absolute_error(y_test, predictions_xgb)
r2_xgb= r2_score(y_test, predictions_xgb)

print('Mean Squared Error:', mse_xgb)
print('Root Mean Squared Error:', rmse_xgb)
print('Mean Absolute Error:', mae_xgb)
print('R² Score:', r2_xgb)


Mean Squared Error: 0.4703763118605287
Root Mean Squared Error: 0.6798005046250898
Mean Absolute Error: 0.5156661958161896
R² Score: 0.9738799745071066


## PREDICTION HUMIDITY 
### linear regression

In [40]:
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
import pandas as pd

# Divisez les données en ensembles d'entraînement et de test
X1_train = df.drop(['Date/Time','target_hum','target_tmp','target_vent'], axis=1).iloc[:7991]  # Premières 7991 lignes pour l'entraînement
y1_train = df['target_hum'].iloc[:7991]

X1_test = df.drop(['Date/Time','target_hum','target_tmp','target_vent'], axis=1).iloc[-768:]  # Dernières 768 lignes pour le test
y1_test = df['target_hum'].iloc[-768:]

# Choisissez le modèle de régression (dans cet exemple, RandomForestRegressor)
model1 =LinearRegression()

# Entraînez le modèle sur les données d'entraînement
model1.fit(X1_train, y1_train)

# Faites des prédictions sur les données de test
predictionss = model1.predict(X1_test)

# Évaluez les performances du modèle
mse1 = mean_squared_error(y1_test, predictionss)
rmse1 = mse1 ** 0.5  # Root Mean Squared Error
mae1 = mean_absolute_error(y1_test, predictionss)
r21 = r2_score(y1_test, predictionss)

print('Mean Squared Error:', mse1)
print('Root Mean Squared Error:', rmse1)
print('Mean Absolute Error:', mae1)
print('R² Score:', r21)

# Afficher les prédictions et les valeurs réelles dans un DataFrame
predictions_df1 = pd.DataFrame({'Prédictions': [round(pred, 1) for pred in predictionss], 
                               'Humidité Réelle': [round(real, 1) for real in y1_test]})
print(predictions_df1)


Mean Squared Error:

 10.274620299186747
Root Mean Squared Error: 3.205404857297553
Mean Absolute Error: 2.334020433218177
R² Score: 0.9122282136945584
     Prédictions  Humidité Réelle
0           64.0             69.0
1           69.0             76.0
2           75.8             71.0
3           71.0             68.0
4           68.2             75.0
..           ...              ...
763         79.9             83.0
764         82.0             93.0
765         91.5             89.0
766         87.6             86.0
767         84.7             86.0

[768 rows x 2 columns]


## PREDICTION VENT
### linear regression

In [41]:
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score

# Charger les données
X2_train = df.drop(['Date/Time','target_vent','target_tmp','target_hum'], axis=1).iloc[:7991]
y2_train = df['target_vent'].iloc[:7991]

X2_test = df.drop(['Date/Time','target_vent','target_tmp','target_hum'], axis=1).iloc[-768:]
y2_test = df['target_vent'].iloc[-768:]

model2 =LinearRegression()

model2.fit(X2_train, y2_train)

# Faire des prédictions sur l'ensemble de test
predictions2 = model2.predict(X2_test)

# Calculer les métriques de régression
mse2 = mean_squared_error(y2_test, predictions2)
rmse2 = mse2 ** 0.5  # Root Mean Squared Error
mae2 = mean_absolute_error(y2_test, predictions2)
r22 = r2_score(y2_test, predictions2)

# Afficher les métriques de régression
print('Mean Squared Error:', mse2)
print('Root Mean Squared Error:', rmse2)
print('Mean Absolute Error:', mae2)
print('R² Score:', r22)


Mean Squared Error: 20.57607399249657
Root Mean Squared Error: 4.536085756739677
Mean Absolute Error: 3.438817952332263
R² Score: 0.761139098876764


In [42]:
# Afficher les prédictions et les valeurs réelles
predictions2 = pd.DataFrame({'Prédictions': predictions2, 'vent speed': y2_test})
print(predictions2)


      Prédictions  vent speed
8016    18.366262         9.0
8017     9.969993         6.0
8018     7.481108         4.0
8019     5.769127         4.0
8020     5.753750         6.0
...           ...         ...
8779    27.690515        24.0
8780    22.645419        28.0
8781    26.066598        28.0
8782    26.046011        30.0
8783    27.715553        30.0

[768 rows x 2 columns]


In [43]:
# Convertir les prédictions en entiers
predictions_int = predictions2.astype(int)

# Afficher le DataFrame avec les prédictions converties en entiers
print(predictions_int)


      Prédictions  vent speed
8016           18           9
8017            9           6
8018            7           4
8019            5           4
8020            5           6
...           ...         ...
8779           27          24
8780           22          28
8781           26          28
8782           26          30
8783           27          30

[768 rows x 2 columns]


## predict all hours of the day

In [44]:
import numpy as np

prediction_tmp = np.array([17])
prediction_hum=np.array([79.])
prediction_vent=np.array([18.])


df_new1 = {
    'Date/Time':'01:00',
    
    'Temp_C': prediction_tmp[0],
    'Rel Hum_%': prediction_hum[0],
    'Wind Speed_km/h': prediction_vent[0]
}
df_new1 = pd.DataFrame(df_new1, index=[1])

print(df_new1)


  Date/Time  Temp_C  Rel Hum_%  Wind Speed_km/h
1     01:00      17       79.0             18.0


In [45]:
import pandas as pd
import numpy as np

# Supposons que new_data est votre dataframe d'origine
new_data = pd.DataFrame({
    'Date/Time': ['00:00'],
    'Temp_C': [7],
    'Rel Hum_%': [79.0],
    'Wind Speed_km/h': [18.0]
})

# Nombre d'heures à prédire
hours_to_predict = 23

# Créer un dataframe pour stocker toutes les prédictions
all_predictions = pd.DataFrame(columns=['Date/Time', 'Temp_C', 'Rel Hum_%', 'Wind Speed_km/h'])
all_predictions = pd.concat([all_predictions, new_data], ignore_index=True)

# Copier les données d'origine pour le processus de prédiction
data = new_data.copy()

for hour in range(1, hours_to_predict + 1):  
    # Sélectionner les données pour les 24 dernières heures
    data_new = data.drop(['Date/Time'], axis=1)

    # Prédire les valeurs pour la prochaine heure
    prediction_vent = int(model2.predict(data_new)[0])
    prediction_hum = int(model1.predict(data_new)[0])
    prediction_tmp = round(model.predict(data_new)[0], 1)
    
    # Stocker les prédictions dans un dataframe temporaire
    temp_df = pd.DataFrame({
        'Date/Time': [f'{hour:02d}:00'],
        'Temp_C': prediction_tmp,
        'Rel Hum_%': prediction_hum,
        'Wind Speed_km/h': prediction_vent
    })
    
    # Ajouter la nouvelle ligne à all_predictions
    all_predictions = pd.concat([all_predictions, temp_df], ignore_index=True)

    # Mettre à jour les données d'entrée pour la prochaine prédiction
    data = pd.concat([data, temp_df], ignore_index=True)
    data = data.iloc[1:]

# Afficher toutes les prédictions
print(all_predictions)


C:\Users\User\AppData\Local\Temp\ipykernel_3112\4059853622.py:17: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  all_predictions = pd.concat([all_predictions, new_data], ignore_index=True)


   Date/Time Temp_C  Rel Hum_%  Wind Speed_km/h
0      00:00      7       79.0             18.0
1      01:00    7.0       78.0             17.0
2      02:00    7.1       77.0             16.0
3      03:00    7.2       76.0             15.0
4      04:00    7.3       75.0             15.0
5      05:00    7.3       74.0             15.0
6      06:00    7.3       73.0             15.0
7      07:00    7.3       72.0             14.0
8      08:00    7.3       71.0             14.0
9      09:00    7.3       70.0             14.0
10     10:00    7.3       69.0             14.0
11     11:00    7.3       68.0             14.0
12     12:00    7.3       67.0             14.0
13     13:00    7.3       66.0             14.0
14     14:00    7.3       66.0             14.0
15     15:00    7.3       66.0             14.0
16     16:00    7.3       66.0             14.0
17     17:00    7.3       66.0             14.0
18     18:00    7.3       66.0             14.0
19     19:00    7.3       66.0          

In [46]:
from pymongo import MongoClient
import pandas as pd
client = MongoClient("mongodb://localhost:27017/")
db = client["station_meteorologique"]
collection = db["predict all hours"]
for index, row in all_predictions.iterrows():
    prediction_doc = {
        "Date/Time": row["Date/Time"],
        "Temp_C": row["Temp_C"],
        "Rel Hum_%": row["Rel Hum_%"],
        "Wind Speed_km/h": row["Wind Speed_km/h"],
    }

    # Insérer le document de prédiction dans la collection MongoDB
    insert_result = collection.insert_one(prediction_doc)

    # Vérifier le résultat de l'insertion
    if insert_result.acknowledged:
        print(f"Prédiction pour Date/Time {row['Date/Time']} insérée avec succès dans MongoDB.")
    else:
        print(f"Erreur lors de l'insertion de la prédiction pour Date/Time {row['Date/Time']} dans MongoDB.")
        print("Détails de l'erreur:", insert_result.raw_result)

Prédiction pour Date/Time 00:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 01:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 02:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 03:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 04:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 05:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 06:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 07:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 08:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 09:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 10:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 11:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 12:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 13:00 insérée avec succès dans MongoDB.
Prédiction pour Date/Time 14:00 insérée avec succès dans MongoDB.
Prédiction